instrumentation/spring/spring-webmvc-3.1/wildfly-testing/src/test/resources/web.xml

```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<web-app xmlns="http://xmlns.jcp.org/xml/ns/javaee"
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://xmlns.jcp.org/xml/ns/javaee
+         http://xmlns.jcp.org/xml/ns/javaee/web-app_3_1.xsd"
+  version="3.1">
+
+  <servlet>
+    <servlet-name>dispatcher</servlet-name>
+    <servlet-class>org.springframework.web.servlet.DispatcherServlet</servlet-class>
+    <load-on-startup>1</load-on-startup>
+  </servlet>
+
+  <!--
+    Spring boot automatically registers a servlet filter when spring bean implements
+    javax.servlet.Filter but vanilla spring does not so we need to add a placeholder
+    for our servlet filter here.
+  -->
+  <filter>
+    <filter-name>otelAutoDispatcherFilter</filter-name>
+    <filter-class>org.springframework.web.filter.DelegatingFilterProxy</filter-class>
+  </filter>
+
+  <filter>
+    <filter-name>testFilter</filter-name>
+    <filter-class>org.springframework.web.filter.DelegatingFilterProxy</filter-class>
+  </filter>
```

Comment:
it's not clear to me why the double registration?

Reply:
Because there are 2 filters `otelAutoDispatcherFilter` is used by the filter added by instrumentation, `testFilter` is used by test. Basically what the integration does it attempts to figure out what would handle the request before the request processing reaches spring so to test it we need something that also runs before spring request processing to handle it so that it never reaches spring. I guess this could be useful when there is a spring security filter that returns not authorized, without this integration there wouldn't be a nice path for such requests.
With spring boot `otelAutoDispatcherFilter` would be added automatically but not with plain spring so this instrumentation is kind of useless when not using spring boot.

Reply:
thx, I finally got it 😅